subprojects/robotpy-wpilib/wpilib/iterativerobotpy.py

```diff
@@ -0,0 +1,476 @@
+            print(f"Default autonomousPeriodic() method... Override me!")
+            self._autonomousPeriodicHasRun = True
+
+    def teleopPeriodic(self) -> None:
+        """
+        Periodic code for teleop mode should go here.
+
+        Users should override this method for code which will be called each time a
+        new packet is received from the driver station and the robot is in teleop
+        mode.
+        """
+        if not self._teleopPeriodicHasRun:
+            print(f"Default teleopPeriodic() method... Override me!")
+            self._teleopPeriodicHasRun = True
+
+    def testPeriodic(self) -> None:
+        """
+        Periodic code for test mode should go here.
+
+        Users should override this method for code which will be called each time a
+        new packet is received from the driver station and the robot is in test
+        mode.
+        """
+        if not self._testPeriodicHasRun:
+            print(f"Default testPeriodic() method... Override me!")
+            self._testPeriodicHasRun = True
+
+    def disabledExit(self) -> None:
+        """
+        Exit code for disabled mode should go here.
+
+        Users should override this method for code which will be called each time
+        the robot exits disabled mode.
+        """
+        pass
+
+    def autonomousExit(self) -> None:
+        """
+        Exit code for autonomous mode should go here.
+
+        Users should override this method for code which will be called each time
+        the robot exits autonomous mode.
+        """
+        pass
+
+    def teleopExit(self) -> None:
+        """
+        Exit code for teleop mode should go here.
+
+        Users should override this method for code which will be called each time
+        the robot exits teleop mode.
+        """
+        pass
+
+    def testExit(self) -> None:
+        """
+        Exit code for test mode should go here.
+
+        Users should override this method for code which will be called each time
+        the robot exits test mode.
+        """
+        pass
+
+    def setNetworkTablesFlushEnabled(self, enabled: bool) -> None:
+        """
+        Enables or disables flushing NetworkTables every loop iteration.
+        By default, this is enabled.
+
+        :deprecated: Deprecated without replacement.
+
+        :param enabled: True to enable, false to disable.
+        """
+
+        self._ntFlushEnabled = enabled
+
+    def enableLiveWindowInTest(self, testLW: bool) -> None:
+        """
+        Sets whether LiveWindow operation is enabled during test mode.
+
+        :param testLW: True to enable, false to disable. Defaults to false.
+                       @throws if called in test mode.
+        """
+        if self.isTestEnabled():
+            raise RuntimeError("Can't configure test mode while in test mode!")
+        if not self._reportedLw and testLW:
+            report(_kResourceType_SmartDashboard, _kSmartDashboard_LiveWindow)
```

Comment:
This method shouldn't be on the critical path for the control loop, so no need to try to optimise accessing globals here.